tests/cpp/test_topk_device_func.cpp

```diff
@@ -0,0 +1,325 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <gtest/gtest.h>
+
+#include <tests/cpp/topk_test_helper.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <algorithm>
+#include <cstdint>
+#include <random>
+#include <vector>
+
+namespace nvfuser {
+
+using TopkDeviceFuncTest = NVFuserTest;
+
+template <typename DataT>
+std::vector<DataT> getVector(at::Tensor tensor) {
+  NVF_ERROR_EQ(tensor.dim(), 1);
+  if (tensor.dtype() == at::kBFloat16) {
+    tensor = tensor.to(at::kFloat);
+  }
+  auto cpu_tensor = tensor.cpu();
+  auto total_elements = tensor.size(0);
+  return std::vector<DataT>(
+      cpu_tensor.data_ptr<DataT>(),
+      cpu_tensor.data_ptr<DataT>() + total_elements);
+}
+
+// Helper function to validate topk correctness
+template <typename DataT>
+bool validateTopkOrder(
+    const at::Tensor& input_tensor,
+    const at::Tensor& values_tensor,
+    const at::Tensor& indices_tensor,
+    int64_t k,
+    bool largest = true) {
+  NVF_ERROR_EQ(cudaDeviceSynchronize(), cudaSuccess);
+
+  auto input_data = getVector<DataT>(input_tensor);
+  auto output_values = getVector<DataT>(values_tensor);
+  auto output_indices = getVector<int64_t>(indices_tensor);
+
+  // Check that we have k valid results
+  if (static_cast<int64_t>(output_values.size()) < k ||
+      static_cast<int64_t>(output_indices.size()) < k) {
+    return false;
+  }
+
+  // Check valid indices range
+  for (int64_t i = 0; i < k; i++) {
+    if (output_indices[i] < 0 || output_indices[i] >= input_data.size()) {
```

Comment:
```suggestion
    if (output_indices[i] < 0 || output_indices[i] >= static_cast<int64_t>(input_data.size())) {
```
**[Suggested by AI]** The code change addresses a potential issue with comparing a signed integer (`output_indices[i]`) to an unsigned integer (`input_data.size()`).  Specifically, it explicitly casts `input_data.size()` to a signed `int64_t` before the comparison.

**Summary:**

The change prevents a potential out-of-bounds check failure due to implicit type conversion during comparison.  `input_data.size()` likely returns an unsigned type (like `size_t`), and comparing a negative `output_indices[i]` (which is an `int`) directly to it can lead to unexpected behavior.  The cast ensures both operands are the same signed type, guaranteeing the comparison works as intended and prevents false positives in the bounds check. This fixes a build error likely caused by this comparison.